### DISCUSSION TITLE
Get old data OnRecordAfterUpdateRequest Hook

### DISCUSSION THREAD

**Discussion by iamdineshkumar** [pocketbase/pocketbase#2257] [Q&A] [answered]
I have a invoice as collection and invoice items to hold all invoice items. While invoice gets edited user may change invoice items itself by adding or removing items. So that I need to remove items which are not mapped after updation of invoice. If there is old data i can compare and procced. How to get old data after updare hook.

**Comment by ganigeorgiev** [comment, marked as answer]
> How to get old data after updare hook.

You can access the original/old record using `e.Record.OriginalCopy()`, but note that an after* hook may not be the correct hook for what you are trying to do. If you want to intercept user submitted data the before* hooks would be better since the record would not be persisted yet and you'll be able to return an error to prevent data change.

**Reply by iamdineshkumar** [reply]
I have planned to insert new item and delete all old items. e.Record.OriginalCopy().GetString() will return only one relation right? But there are multiple items under invoice. How to handle it to delete all records from invoice items collection?

**Reply by ganigeorgiev** [reply]
If you want to get a slice of string, you can use the `GetStringSlice()` helper (for other common `Record` helpers you can check https://pocketbase.io/docs/record-methods/#getset-record-fields).

**Reply by iamdineshkumar** [reply]
In case I use OnRecordBeforeUpdateRequest hook and did some record changes in othere collection but it failes due to any reason whether all changes will be reversed?
